acme/acme/fields.py

```diff
@@ -78,5 +84,13 @@ def rfc3339(json_name: str, omitempty: bool = False) -> Any:
 
 
 def resource(resource_type: str) -> Any:
-    """Generates a type-friendly Resource field."""
-    return Resource(resource_type)
+    """Generates a type-friendly Resource field.
+
+    .. deprecated: 1.30.0
+
+    """
+    warnings.warn('acme.fields.resource is deprecated and will be removed soon.',
+                    DeprecationWarning, stacklevel=2)
+    with warnings.catch_warnings():
+        warnings.filterwarnings('ignore', '.*Resource', DeprecationWarning)
+        return Resource(resource_type)
```

Comment:
This is maybe a nitpick, but instead of warning in `__init__` and the body of the `resource` function, can we warn when these attributes are accessed in the module like you did in `messages.py`? I suppose I'm less worried about the `resource` function, but I personally think it's nice for deprecated classes.